server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlVisitorHelper.java

```diff
@@ -150,4 +157,42 @@ public static Position getAdjustedEndPosition(
     }
     return end;
   }
+
+  public static ExecSqlWheneverNode.WheneverConditionType getConditionType(Db2SqlParser.Dbs_wheneverContext ctx) {
+    ParserRuleContext ruleContext = ((ParserRuleContext) ctx);
+    ExecSqlWheneverNode.WheneverConditionType conditionType = ExecSqlWheneverNode.WheneverConditionType.NOT_FOUND;
+
+
+    if (ruleContext.getChildCount() >= 3) {
+      ParseTree pt = ruleContext.getChild(1);
+      String value = pt.getText().trim().toUpperCase();
+
+      if (Objects.equals(value, "SQLERROR")) {
+        conditionType = ExecSqlWheneverNode.WheneverConditionType.SQLERROR;
+      } else if (Objects.equals(value, "SQLWARNING")) {
+        conditionType = ExecSqlWheneverNode.WheneverConditionType.SQLWARNING;
+      }
+    }
+    return conditionType;
```

Comment:
Now that this is being moved, it could be simplified as well.
```suggestion
    if (ctx.SQLERROR() != null)
        return ExecSqlWheneverNode.WheneverConditionType.SQLERROR;
    else if (ctx.SQLWARNING() != null)
        return ExecSqlWheneverNode.WheneverConditionType.SQLWARNING;
    else
        return ExecSqlWheneverNode.WheneverConditionType.NOT_FOUND;
```